.github/workflows/osx_testing.yml

```diff
@@ -0,0 +1,185 @@
+          # Must be revisited after fixing https://github.com/tarantool/tarantool/issues/6477
+          #
+          # Added -DENABLE_DIST=ON in case of tarantoolctl abcense in destdir. 
+          # Fixes not-brew tarantool build on Mac OS
+          cmake . -DENABLE_DIST=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl@1.1 -DOPENSSL_LIBRARIES=/usr/local/opt/openssl@1.1/lib
+          # {{{ Workaround Mac OS build failure (gh-6076)
+          #
+          # https://github.com/tarantool/tarantool/issues/6076
+          #
+          # In brief: when "src/lib/small" is in include paths,
+          # `#include <version>` from inside Mac OS SDK headers
+          # attempts to include "src/lib/small/VERSION" as a
+          # header file that leads to a syntax error.
+          #
+          # It was fixed in the following commits:
+          #
+          # * 1.10.10-24-g7bce4abd1
+          # * 2.7.2-44-gbb1d32903
+          # * 2.8.1-56-ga6c29c5af
+          # * 2.9.0-84-gc5ae543f3
+          #
+          # However applying the workaround for all versions looks
+          # harmless.
+          #
+          # Added -f just in case: I guess we'll drop this useless
+          # obsoleted VERSION file from the git repository sooner
+          # or later.
+          rm -f src/lib/small/VERSION
+          # The same as above, but for the VERSION file generated
+          # by tarantool's CMake script.
+          rm VERSION
+          # }}} Workaround Mac OS build failure (gh-6076)
+          # Continue the build.
+          make -j$(sysctl -n hw.logicalcpu)
+          make DESTDIR="${T_DESTDIR}" install
+        if: matrix.tarantool != 'brew' && steps.cache.outputs.cache-hit != 'true'
+
+      - name: Export TARANTOOL_DIR and PATH
+        run: |
+          printf '%s=%s\n' TARANTOOL_DIR "${T_DESTDIR}/usr/local" >> "${GITHUB_ENV}"
+          printf '%s\n' "${T_DESTDIR}/usr/local/bin" >> "${GITHUB_PATH}"
+        if: matrix.tarantool != 'brew'
+
+      - name: Verify tarantool version
+        run: |
+          # Workaround https://github.com/tarantool/tarantool/issues/4983
+          # Workaround https://github.com/tarantool/tarantool/issues/5040
+          tarantool -e "require('fiber').sleep(0) assert(_TARANTOOL:startswith('${T_VERSION}'), _TARANTOOL) os.exit()"
+        if: matrix.tarantool != 'brew' && matrix.tarantool != 'master'
+
+      - name: Clone the connector
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.SRCDIR }}
+          # Needed because of tarantool-c dependence on submodules
+          submodules: recursive
+          fetch-depth: 0
+
+      - name: Build the connector
+        run: |
+          cd "${SRCDIR}"
+          cmake . && make
+
+      - name: Run tests
+        run: |
+          cd "${SRCDIR}"
+          # Needed to install and use gevent
+          pip3 install -r test-run/requirements.txt
```

Comment:
I would move it to its own step like 'Install testing dependencies'.

Reply:
Shall I add 'ulimit -n 4096' to new step? It seems appropriate.

Reply:
I would keep it in the step, where we run tests.